akka-serialization-jackson/src/main/scala/akka/serialization/jackson/internal/AkkaJacksonSerializationFeature.scala

```diff
@@ -28,28 +37,129 @@ final class AkkaJacksonSerializationFeature extends Feature {
 
-    access.registerSubtypeReachabilityHandler(
-      (_, subtype) =>
-        if (subtype != null && !subtype.isInterface) {
-          log("Registering CborSerializable class: " + subtype.getName)
-          RuntimeReflection.register(subtype)
-          RuntimeReflection.register(subtype.getDeclaredConstructors: _*)
-          RuntimeReflection.register(subtype.getDeclaredFields: _*)
-        },
-      cborSerializable)
+    access.registerSubtypeReachabilityHandler({ (_, subtype) =>
+      if (subtype != null) {
+        registerTypeForJacksonSerialization(access, subtype)
+      }
+    }, cborSerializable)
+
+    val jsonStdSerializer = access.findClassByName(classOf[StdSerializer[_]].getName)
+    access.registerSubtypeReachabilityHandler({ (_, subtype) =>
+      if (subtype != null) {
+        registerCustomJacksonSerializers(subtype)
+      }
+    }, jsonStdSerializer)
+
+    val jsonStdDeserializer = access.findClassByName(classOf[StdDeserializer[_]].getName)
+    access.registerSubtypeReachabilityHandler({ (_, subtype) =>
+      if (subtype != null) {
+        registerCustomJacksonSerializers(subtype)
+      }
+    }, jsonStdDeserializer)
+
+  }
+
+  private def registerTypeForJacksonSerialization(access: Feature.BeforeAnalysisAccess, clazz: Class[_]): Unit = {
+
+    if (!alreadyRegisteredType.contains(clazz.getName) && clazz.getPackage == null || (!clazz.getPackage.getName
+          .startsWith("java") && !clazz.getPackage.getName.startsWith("scala"))) {
+      alreadyRegisteredType.add(clazz.getName)
+      log("Registering for jackson serialization: " + clazz.getName)
+      RuntimeReflection.register(clazz)
+      RuntimeReflection.registerAllDeclaredFields(clazz)
+      RuntimeReflection.registerAllDeclaredMethods(clazz)
+      try {
+        val scalaModuleField = clazz.getDeclaredField("MODULE$")
+        RuntimeReflection.register(scalaModuleField)
+        log("Registering Scala module for " + clazz.getName)
+      } catch {
+        case _: NoSuchFieldException => // no scala module, this is ok
+      }
+
+      // getters
+      val scalaCaseClass = try {
+        clazz.getMethod("productElementNames")
+        true
+      } catch {
+        case _: NoSuchMethodException => false
+      }
+      util.Arrays
+        .stream(clazz.getDeclaredMethods)
+        .forEach(method =>
+          if (method.getParameterTypes.length == 0 && isPossiblyGetter(scalaCaseClass, method.getName)) {
+            log("Registering method " + clazz + "." + method.getName)
+            RuntimeReflection.register(method)
+            RuntimeReflection.registerAsQueried(method)
+          })
+
+      RuntimeReflection.registerAllConstructors(clazz)
+      RuntimeReflection.registerAllDeclaredConstructors(clazz)
+      // we still need explicit register of each constructor for some reason
+      util.Arrays.stream(clazz.getDeclaredConstructors).forEach { constructor =>
+        // FIXME this could probably be more selective
```

Comment:
But unclear to me how Jackson chooses what constructor to use in the face of several, perhaps based on the incoming data and we really need to list all?